### PR TITLE
#160753071 Users should be able to get likes information from an article

### PR DESCRIPTION
**Pull request by JCanaks**
#### What does this PR do?
This enable users to receive the likes information about an article when all `get article` routes are called. It also enables the user to access likes information about an article prior to login.
#### Description of Task to be completed?
- Update the get article and get all article routes to return information about likes for that article
- Show the number of likes for a particular article in all get article responses
- Show an array of users who liked a particular article in all get article responses
- Update the get likes route to return likes information for an article without authentication
#### How should this be manually tested?
- Clone the repository
- Run npm install to install all app dependencies
- Setup postgres (if not done already)
- Create a database on postgres
- Run sequelize db:migrate
- Run npm start to start the app
- Test the following routes
  - GET api/articles - to get all articles and view the corresponding likes for each article
  - GET api/articles/:slug - to get a particular article and view the likes for that article
  - GET api/articles/:slug/like - to get like details of an article including user information about those who liked the article
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#160753071
#### Screenshots (if appropriate)
Get likes route (for an article)
screen shot 2018-09-25 at 1 32 39 pm
Get all articles route
screen shot 2018-09-25 at 1 33 07 pm
Get an article route
screen shot 2018-09-25 at 1 33 36 pm
#### Questions:
N/A

### CODE REVIEW

**Review by mppdg — APPROVED** [review]

File: server/controllers/ArticleController.js [hunk 1]
```diff
@@ -98,6 +99,7 @@ class ArticleController {
       const articleObj = article.dataValues;
       articleObj.tags = articleControllerHelper
         .formatTagResponse(article.tags);
+      articleObj.likesCount = articleObj.likes.length;
```
Comment (rhenshaw56) [inline server/controllers/ArticleController.js:102]: I thought the database had a method for doing the count ? can you check, if it does let's use that instead

File: server/test/routes/likes.test.js [hunk 2]
```diff
@@ -53,30 +53,15 @@ describe('Testing likes routes /api/auth/:slug/like', () => {
...
+  
   describe('when like informations are retrieved when no one has liked the article', () => {
     it('should return a status code of 204 ', (done) => {
       request(server)
         .get(likeRoute)
         .set('Authorization', token)
         .end((err, res) => {
           expect(res)
-            .property('status').to.equal(204);
+            .property('status').to.equal(200);
```
Comment (rhenshaw56) [inline server/test/routes/likes.test.js:64]: the it block says `should return a status code of 204` but the assertion is for 200, why ?

**Review by rhenshaw56 — CHANGES_REQUESTED** [review]

File: server/test/routes/likes.test.js (hunk 2 above)
Comment (JCanaks) [reply server/test/routes/likes.test.js:64]: I will edit the it block statement it should be 200 not 204 following our convention of returning 200 status for get routes even for empty results. I made changes to that effect in the like controller above

File: server/controllers/ArticleController.js (hunk 1 above)
Comment (JCanaks) [reply server/controllers/ArticleController.js:102]: Do you mean getting the count with sequelize? cos i tried doing something of this nature `attributes: [[sequelize.fn('COUNT', sequelize.col('ArticleLike.id')), 'likesCount']]` but was getting some errors so i had to resort to using this method. There seems to be some issues using the count fn when querying certain associated tables

**Pull request merged**